core/Enclave/enclave_hsm.cpp

```diff
@@ -144,6 +135,151 @@ static uint32_t sgx_get_gcm_ciphertext_size(const sgx_aes_gcm_data_ex_t *gcm_dat
     return gcm_data->ciphertext_size;
 }
 
+static uint32_t sgx_get_key_size(const ehsm_keyspec_t keyspec)
+{       
+    switch (keyspec)
+    {
+        case EH_AES_GCM_128:
+            return 16;
+        case EH_AES_GCM_256:
+            return 32; 
+    }
+
+    return UINT32_MAX;
+}
+
+static const EVP_CIPHER * sgx_get_openssl_api(const ehsm_keyspec_t keyspec)
+{       
+    switch (keyspec)
+    {
+        case EH_AES_GCM_128:
+            return EVP_aes_128_gcm();
+        case EH_AES_GCM_256:
+            return EVP_aes_256_gcm();
+    }
```

Comment:
return anything here?

Reply:
return NULL if none keyspec is matched.

Reply:
It should be add a default, return NULL

Reply:
explicitly return null but not depend on compiler.

Reply:
Ok I will return NULL for default.